vllm/v1/worker/gpu_model_runner.py

```diff
@@ -1163,6 +1195,32 @@ def execute_model(
             return hidden_states
 
         sample_hidden_states = hidden_states[logits_indices]
+
+        if self.input_batch.pooling_params:
+            assert self.input_batch.num_reqs ==\
+                 len(self.input_batch.pooling_params), \
+            "Either all or none of the requests in" \
+            " a batch must be pooling request"
+
+            pooler_output = self.model.pooler(
+                hidden_states=hidden_states,
+                pooling_metadata=self.input_batch.pooling_metadata)
+
+            # any token will do because max tokens is 1
+            sampled_tokens = [[0]] * self.input_batch.num_reqs
+
+            return ModelRunnerOutput(
+                req_ids=self.input_batch.req_ids,
+                req_id_to_index=self.input_batch.req_id_to_index,
+                sampled_token_ids=sampled_tokens,
+                spec_token_ids=None,
+                logprobs=None,
+                prompt_logprobs_dict={},
+                pooler_output=[
+                    o.data.to("cpu") for o in pooler_output.outputs
+                ],
```

Comment:
This loop looks very inefficient because CPU-GPU synchronization happens inside the loop. Can we copy the whole tensor at once?